dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/bootstrap/DubboBootstrap.java

```diff
@@ -1103,26 +1108,25 @@ public DubboBootstrap start() {
             }
 
             referServices();
-            if (asyncExportingFutures.size() > 0 || asyncReferringFutures.size() > 0) {
-                new Thread(() -> {
+
+            // wait async export / refer finish if needed
+            awaitFinish();
+
+            new Thread(() -> {
```

Comment:
Is it necessary to create a thread even if users not specify async export?

Reply:
Changed